### PR TITLE
Topic algorithms

### PR DESCRIPTION
**Pull request by satrfall6**
Hi I am Weichung Lai student No. 45033027
this is denoise algorithm 

thanks

### CODE REVIEW

File: denoise/driver copy.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
```
Comment (shakes76) [inline denoise/driver copy.py]: Header block should have more info. Block would include authorship, license, copyright, dates and description of what the driver does.

File: denoise/torch_denoise_tv_chambolle copy.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+       # e.g. gt[0] is the first order finite difference along axis 0
+       slices_gt = [slice(None), ] * (ndim + 1)
+       for ax in range(ndim):
+           slices_gt[ax+1] = slice(0, -1)
+           slices_gt[0] = ax
+           gt[tuple(slices_gt)] = diff(out, ax)
+           slices_gt[ax+1] = slice(None)
+            
+       norm = torch.sqrt((gt ** 2).sum(axis=0)).unsqueeze(0)
+       Et = Et + weight * norm.sum()
```
Comment (shakes76) [inline denoise/torch_denoise_tv_chambolle copy.py]: Good to add inline comments even if original code didn't have them

File: denoise/torch_denoise_tv_chambolle copy.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        Denoised image.
+    
+    """
+    im_type = (image.numpy()).dtype
+    if not im_type.kind == 'f':
+        image = image.type(torch.float64)
+        image = image/torch.abs(image.max()+image.min())
+        
+    if multichannel:
+        out = torch.zeros_like(image)
```
Comment (shakes76) [inline denoise/torch_denoise_tv_chambolle copy.py]: Good to add inline comments even if original code didn't have them

File: denoise/README.md [hunk 4]
```diff
@@ -1,14 +1,70 @@
-# Denoising
-Various denoise algorthms implemented for Tensorflow
```
Comment (shakes76) [inline denoise/README.md]: Please create a directory for your algorithm and move files there. It currently overwrites the denoise Readme.

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
 Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present
## Commit Log
Meaningful commit messages 
Progressive commits usage
## Documentation
Description and Comments good 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time) Folder not used.
Feedback  required. Move files into folder and remove unnecessary files.
Request Description  minimal. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info. Docstrings are important too.

**Comment by satrfall6** [comment]
Hi Shakes 

I have removed unnecessary files and create a new folder to put my files in, so it won't cover the other README. The other change is updating the information in the driver and function. 
Thanks

**Pull request merged**